build.sbt

```diff
@@ -1335,18 +1335,51 @@ lazy val scalaStyleSettings = Seq(
  ****************************
  */
 
+lazy val compileJavastyle = taskKey[Unit]("compileJavastyle")
+lazy val testJavastyle = taskKey[Unit]("testJavastyle")
+
 def javaCheckstyleSettings(checkstyleFile: String): Def.SettingsDefinition = {
   // Can be run explicitly via: build/sbt $module/checkstyle
   // Will automatically be run during compilation (e.g. build/sbt compile)
   // and during tests (e.g. build/sbt test)
   Seq(
     checkstyleConfigLocation := CheckstyleConfigLocation.File(checkstyleFile),
-    checkstyleSeverityLevel := CheckstyleSeverityLevel.Error,
-    (Compile / compile) := ((Compile / compile) dependsOn (Compile / checkstyle)).value,
-    (Test / test) := ((Test / test) dependsOn (Test / checkstyle)).value
+    checkstyleSeverityLevel := CheckstyleSeverityLevel.Ignore,
+
+    compileJavastyle := {
+      (Compile / checkstyle).value
+      javaCheckstyle(streams.value.log, checkstyleOutputFile.value)
+    },
+    (Compile / compile) := ((Compile / compile) dependsOn compileJavastyle).value,
+
+    testJavastyle := {
+      (Test / checkstyle).value
+      javaCheckstyle(streams.value.log, (Compile / target).value / "checkstyle-test-report.xml")
+    },
+    (Test / test) := ((Test / test) dependsOn (Test / testJavastyle)).value
   )
 }
 
+def javaCheckstyle(log: Logger, reportFile: File): Unit = {
+  val report = scala.xml.XML.loadFile(reportFile)
+
+  val errors = (report \\ "file").flatMap { fileNode =>
+    val file = fileNode.attribute("name").get.head.text
+    (fileNode \ "error").map { error =>
+      val line = error.attribute("line").get.head.text
+      val message = error.attribute("message").get.head.text
+      (file, line, message)
+    }
+  }
+
+  if (errors.nonEmpty) {
+    errors.foreach { case (file, line, message) =>
+      log.error(s"- File: $file, Line: $line, Message: $message")
+    }
+    sys.error(s"Found checkstyle errors")
+  }
```

Comment:
It looks good to me. The only request is, can we print the errors as part of the failure message? That way we don't need to scroll up to find the error.

```
    var errorMsg = "Found checkstyle errors"
    errors.foreach { case (file, line, message) =>
      val lineError = s"File: $file, Line: $line, Message: $message"
      log.error(lineError)
      errorMsg += ("\n" + lineError)
    }
    sys.error(errorMsg + "\n")
```

Reply:
@vkorukanti Thanks for reviewing.